docs/RFCS/select_from_index.md

```diff
@@ -0,0 +1,55 @@
+- Feature Name: select_from_index
+- Status: draft
+- Start Date: 2015-08-10
+- RFC PR:
+- Cockroach Issue: #2046
+
+# Summary
+
+Extend the SELECT syntax to allow an optional index to be specified in
+table references. The current table reference syntax is `table` or
+`database.table`. This syntax would be extended to allow `table.index`
+and `database.table.index`.
+
+# Motivation
+
+As indexes are optional in SQL, it is difficult to know when an index
+is used to satisfy a query which in turn makes it difficult to use the
+existing logic test infrastructure to test correctness of index
+scans. Providing for specification of an index to read from will make
```

Comment:
So is this allowed?

```
create table employees (
  id int primary key
  first_name string
  last_name string
);
create index last_name on employees (last_name);

select first_name, last_name from employees:last_name where last_name='darnell';
```

`first_name` is not in the `last_name` index. We need to be able to test cases like this where the index is used to lead to a full record. I think it would be better to put the index specifier in the where clause. Since `where last_name='darnell'` is already just shorthand for `where employees.last_name='darnell'`, it seems straightforward to use the plain table in the FROM clause and an index in WHERE: `SELECT first_name, last_name FROM employees WHERE employees:last_name.last_name = 'darnell'`. We could place whatever restrictions we want, e.g. if one expression in WHERE uses an index, all expressions on the same table must use the same index.